integration-tests/src/test/groovy/com/okta/sdk/tests/it/SmsTemplateIT.groovy

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.tests.it
+
+import com.okta.sdk.resource.SmsTemplateTranslations
+import com.okta.sdk.resource.template.SmsTemplate
+import com.okta.sdk.resource.template.SmsTemplateType
+import com.okta.sdk.tests.it.util.ITSupport
+import org.testng.annotations.Test
+
+import static com.okta.sdk.tests.it.util.Util.assertPresent
+import static com.okta.sdk.tests.it.util.Util.assertNotPresent
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.*
+
+/**
+ * Tests for /api/v1/templates/sms
+ * @since 2.0.0
+ */
+class SmsTemplateIT extends ITSupport {
+
+    @Test
+    void customTemplatesCrudTest() {
+        def templateName = "template-" + UUID.randomUUID().toString()
+
+        // create translations
+        SmsTemplateTranslations smsTemplateTranslations = client.instantiate(SmsTemplateTranslations)
+        smsTemplateTranslations.put("de", "\${org.name}: ihre bestätigungscode ist \${code}")
+        smsTemplateTranslations.put("it", "\${org.name}: il codice di verifica è \${code}")
+
+        // create template
+        SmsTemplate smsTemplate = client.createSmsTemplate(client.instantiate(SmsTemplate)
+            .setName(templateName)
+            .setType(SmsTemplateType.CODE)
+            .setTemplate("\${org.name}: your verification code is \${code}")
+            .setTranslations(smsTemplateTranslations))
+        registerForCleanup(smsTemplate)
+
+        assertThat(smsTemplate.getId(), notNullValue())
+
+        // list templates
+        assertPresent(client.listSmsTemplates(), smsTemplate)
+
+        // retrieve template
+        SmsTemplate retrievedSmsTemplate = client.getSmsTemplate(smsTemplate.getId())
+        assertThat(retrievedSmsTemplate, notNullValue())
+        assertThat("Incorrect Translations Count", retrievedSmsTemplate.getTranslations().size() == 2)
+
+        // partial update template
+        retrievedSmsTemplate.getTranslations().put("es", "\${org.name}: su código de inscripción es \${code}")
+        retrievedSmsTemplate.partialUpdate()
```

Comment:
`partialUpdate()`?
Is this actually a partial update? The SDK will likely post the full object, so this _should_ be a just `update()` to match all the other resources

Reply:
As per https://developer.okta.com/docs/reference/api/templates/#partial-sms-template-update, any update to _some_ of the properties (in our test case it is `Any translation with non-empty/null value is updated.`) is considered to be a `partial update`.

Reply:
For this test you should probably create a new instance, and just set that property,
It looks like you are reusing the previous instance (which would be the equivalent to a regular update)

Reply:
addressed this.